### DISCUSSION TITLE
stream.on is undefined when using streaming updates

### DISCUSSION THREAD

**Discussion by nguyenlam123** [axios/axios#6144] [General]
I get `stream.on` is `undefined` when using streaming updates. Anyone knows why? I'm using axios v1.6.0

```js
const response = await axios.post('https://stream.example.com', JSON.stringify(data), {
    headers: {Authorization: `Bearer ${token}`, 
    responseType: 'stream'
});

const stream = response.data; // I get a string which represent part of the stream so the stream setup seems to work

stream.on('data', data => { // Any other way I can call the on listener here instead of going through the stream variable? 🤔 
    console.log(data);
});

stream.on('end', () => {
    console.log("stream done");
});
```

Thank you so much in advance for all the feedback! 🙇 

Based my examples on 

- https://github.com/axios/axios#axios-api
- https://stackoverflow.com/questions/71534322/http-stream-using-axios-node-js

**Comment by jasonsaayman** [comment]
`responseType: 'stream'` only returns a Node.js readable stream when Axios is using the Node HTTP adapter.

If you are running this in a browser, React Native, or another environment using the XHR/fetch adapter, `response.data` will not be a Node stream, so `stream.on(...)` will be undefined. In those environments you need to use the platform's streaming APIs instead, such as `fetch()` with `response.body.getReader()` in browsers that support it.

Also, your sample has `responseType` inside `headers`. It should be a top-level Axios config option:

```js
const response = await axios.post('https://stream.example.com', data, {
  responseType: 'stream',
  headers: {
    Authorization: `Bearer ${token}`
  }
});

response.data.on('data', chunk => {
  console.log(chunk);
});

response.data.on('end', () => {
  console.log('stream done');
});
```

So if this code is in Node.js, move `responseType` out of `headers`. If it is in the browser, `stream.on` is not available because that is a Node stream API.
